### ISSUE TITLE
Add ability to rename any unit

### ISSUE THREAD

**Issue by chaorace** [chaorace/cqui#196]
Currently only units promoted beyond a certain point can be renamed, I see no harm in enabling this for any unit.

**Comment by Defectiv74** [comment]
Any unit any time? That would be great!

**Comment by vttale** [comment]
Traders too?  Possible?

**Comment by ricanuck** [comment]
I am trying this, and so far what I see is that there is a call to a function that determines if a given unit CAN be renamed. That function, as far as I know, is very likely inside the dll.

**Comment by chaorace** [comment]
Yep, just checked and the UnitManager doesn't let you issue that command unless it's not already named AND at least level 2. It's out of our hands at the point. Now, there's nothing wrong with adding our own nickname layer on, is that something people will consider worthwhile?

**Comment by Seileach** [comment]
a nickname would be great. i like assigning my units to armies, "palace guard", "northern army" and so on, so a nickname layer would probably do it for me :)

**Comment by JHCD** [comment]
Much work for > 200 military units... :)
